src/librustc/mir/visit.rs

```diff
@@ -527,6 +527,44 @@ macro_rules! make_mir_visitor {
                         );
                     }
 
+                    TerminatorKind::InlineAsm {
+                        template: _,
+                        operands,
+                        options: _,
+                        destination: _,
+                    } => {
+                        for op in operands {
+                            match op {
+                                InlineAsmOperand::In { value, .. }
+                                | InlineAsmOperand::Const { value } => {
+                                    self.visit_operand(value, source_location);
+                                }
+                                InlineAsmOperand::Out { place, .. } => {
+                                    if let Some(place) = place {
+                                        self.visit_place(
+                                            place,
+                                            PlaceContext::MutatingUse(MutatingUseContext::Store),
```

Comment:
There’s a special context `AsmOutput` for outputs. If it is consciously not used, document why. My intuition is saying plain `Store` is valid only for late outputs.

Reply:
`AsmOutput` is only needed for the legacy asm which had read-write operands in MIR. The new asm will lower `inout` operands into an `in_value: Operand` which is read from, and a separate `out_place: Place` which is written to.

Late vs early outputs only affect register allocation. It has no effect on the Rust-level semantics.

Reply:
Oh right, I did indeed miss that particularity entirely.